assets/scripts/store/reducers/ui.js

```diff
@@ -16,6 +17,16 @@ const ui = (state = initialState, action) => {
         ...state,
         streetNameCanvasVisible: false
       }
+    case SET_UNIT_SETTINGS:
+      const imperial = (action.unit === constants.SETTINGS_UNITS_IMPERIAL)
+      return {
+        ...state,
+        unitSettings: {
+          resolution: (imperial) ? constants.SEGMENT_WIDTH_RESOLUTION_IMPERIAL : constants.SEGMENT_WIDTH_RESOLUTION_METRIC,
+          draggingResolution: (imperial) ? constants.SEGMENT_WIDTH_DRAGGING_RESOLUTION_IMPERIAL : constants.SEGMENT_WIDTH_DRAGGING_RESOLUTION_METRIC,
+          clickIncrement: (imperial) ? constants.SEGMENT_WIDTH_CLICK_INCREMENT_IMPERIAL : constants.SEGMENT_WIDTH_CLICK_INCREMENT_METRIC
```

Comment:
Also set the `unitSettings` object in `initialState` -- this way we guarantee that unit settings always has a value and that components like `StreetMetaWidth` doesn't need import its own defaults.

Reply:
@louh Do we set it to imperial values or metric constants?